File: Sorting/LeetCode/sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        vector<char> v;
        for(int i = 0; i < s.size(); i++){
            if(isVowel(s[i])){
                v.push_back(s[i]);
            }
        }
        if(v.size() > 0){
            sort(v.begin(), v.end());
        }
        string ans = "";
        int j = 0;
        for(int i = 0; i < s.size(); i++){
            if(isVowel(s[i])){
                ans += v[j];
                j++;
            }else{
                ans += s[i];
            }
        }
        return ans;
    }
    bool isVowel(char c){
        if(c == 'a' || c == 'e'  || c == 'i' || c == 'o' || c == 'u')
            return true;
        if(c == 'A' || c == 'E'  || c == 'I' || c == 'O' || c == 'U')
            return true;
        return false;
    }
};

File: String/LeetCode/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        String tem = "";
        for(int i = 0; i < s.length(); i++){
            if(isVowel(s.charAt(i))){
                tem += s.charAt(i);
            }
        }
        char[] charArray = tem.toCharArray();
        Arrays.sort(charArray);
        String vs = new String(charArray);
        int j = 0;
        StringBuilder st = new StringBuilder(s); 
        for(int i = 0; i < s.length(); i++){
            if(isVowel(s.charAt(i))){
                st.setCharAt(i, vs.charAt(j));
                j++;
            }
        }
        String rs = st.toString();
        return rs;
    }
    public boolean isVowel(char c){
        c = Character.toLowerCase(c);
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') return true;
        return false;
    }
}